src/frame.js

```diff
@@ -0,0 +1,34 @@
+'use strict';
+
+function Frame() {
+  this.rolls = [];
+  this.score = 0;
+  this.strike = false;
+  this.spare = false;
+}
+
+Frame.prototype.bowl = function(pins) {
+  if (this.score + pins > 10) {
+    throw new Error('Please re-enter correct score');
+  }
+  this.rolls.push(pins);
+  this.score += pins;
+  this.assignBonus();
+};
+
+Frame.prototype.isEndFrame = function() {
+  return (this.rolls.length === 2 || this.score === 10);
+};
+
+Frame.prototype.isStrike = function() {
+  return this.rolls[0] === 10;
+};
+
+Frame.prototype.isSpare = function() {
+  return this.rolls[0] + this.rolls[1] === 10;
+};
+
+Frame.prototype.assignBonus = function() {
```

Comment:
This is pointless.  There is no need to store two versions of the truth.  Since `strike` and `spare` are both derived, they are unnecessary.  If I want to know if a frame is a strike, I can just call `isStrike()`.